app/src/components/nav-menu/UserHomePage.js

```diff
@@ -0,0 +1,223 @@
+import React, {Component} from 'react';
+import {Redirect} from 'react-router-dom';
+import UsersDataRequests from '../../common/UsersDataRequests'
+import endPointsList from '../../common/endPointsList';
+import NavigateMenu from './NavigateMenu'
+import {Validations} from '../../common/Validations';
+
+import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import '../../css/App.css';
+import {connect} from "react-redux";
+
+class UserHomePage extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isEdit: false,
+            isLoggedIn: true,
+            isNeedUserInfo: true,
+            userInfo: {},
+            formattedUserInfo: ''
+        };
+
+        this.loadUserInfo();
+
+        this.onSignOut = this.onSignOut.bind(this);
+        this.onInfoClick = this.onInfoClick.bind(this);
+        this.handleSaveInfoClick = this.handleSaveInfoClick.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(event){
+        const thisInput = {
+            content: '',
+            // isValid: false
+        };
+        // thisInput.isValid = Validations.typeOfRegexp(event.target.value, typeOfRegexp);
+        thisInput.content = event.target.value;
+
+        this.props.changeUserInfoFormInput({key: event.target.name, value: thisInput});
+    }
+
+    onSignOut() {
+        localStorage.removeItem('auth-tok');
+        this.setState({...this.state, isLoggedIn: false});
+    }
+
+    isTokenGood() {
+        const checkToken = localStorage.getItem('auth-tok');
+        if (checkToken) {
+            this.setState({...this.state, isLoggedIn: true});
+        } else {
+            this.setState({...this.state, isLoggedIn: false});
+        }
+    }
+
+    onInfoClick(event) {
+        const input = event.target;
+        const readMode = 'user-page-userInfo-form-read';
+        const editMode = 'user-page-userInfo-form-edit';
+
+        if (input.className.indexOf(editMode) === -1) {
+            input.classList.remove(readMode);
+            input.classList.add(editMode);
+            input.removeAttribute('readonly');
+            this.setState({...this.state, isEdit:true});
+            const saveBtnBlock = document.getElementById('save-btn-block');
+
+            if (!document.getElementById('user-page-saveButton')) {
+                const saveBtn = document.createElement('input');
```

Comment:
Generally, this is not React-like approach. Could you please think through different logic to create new elements only inside render-method